src/sst/core/profile/syncProfileTool.h

```diff
@@ -0,0 +1,111 @@
+// Copyright 2009-2022 NTESS. Under the terms
+// of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Copyright (c) 2009-2022, NTESS
+// All rights reserved.
+//
+// This file is part of the SST software package. For license
+// information, see the LICENSE file in the top level directory of the
+// distribution.
+
+#ifndef SST_CORE_PROFILE_SYNCPROFILETOOL_H
+#define SST_CORE_PROFILE_SYNCPROFILETOOL_H
+
+#include "sst/core/eli/elementinfo.h"
+#include "sst/core/sst_types.h"
+#include "sst/core/ssthandler.h"
+#include "sst/core/warnmacros.h"
+
+#include <chrono>
+#include <map>
+
+namespace SST {
+
+namespace Profile {
+
+// Initial version of sync profiling tool.  The API is not yet complete.
+class SyncProfileTool : public ProfileTool
+{
+public:
+    SST_ELI_REGISTER_PROFILETOOL_DERIVED_API(SST::Profile::SyncProfileTool, SST::Profile::ProfileTool, Params&)
+
+    SST_ELI_DOCUMENT_PARAMS(
+        // { "level", "Level at which to track profile (global, type, component, subcomponent)", "component" },
+        // { "track_ports",  "Controls whether to track by individual ports", "false" },
+        // { "profile_sends",  "Controls whether sends are profiled (due to location of profiling point in the code, turning on send profiling will incur relatively high overhead)", "false" },
+        // { "profile_receives",  "Controls whether receives are profiled", "true" },
```

Comment:
Are these supposed to be commented out?

Reply:
Just to remind me that I need to add Params back in later.  This is just a first cut that I needed in to get some data for a write-up.  It will be extended later on to include all the final functionality.  I really only need the level param, however,  I'll remove the others.